valora-inc/wallet
# feat(keyless-backup): add sign in with google

### Description

Implements button click to initiate google sign in flow. Result saved on redux but not used. Also encrypts .env.* files since they now include client ids. See [this thread](https://valora-app.slack.com/archives/C025V1D6F3J/p1684179136241829) for more context.

To be merged after #3741 

### Test plan

Manually by running the following steps

1. Enable backup locally by changing the feature flag to true (this line needs to change to true: https://github.com/valora-inc/wallet/blob/94fee8a/src/account/Settings.tsx#L374)
2. Build and run app, navigate to settings
3. Click on Keyless backup, click continue, then Sign In With Google and complete sign in
4. Ensure idToken is saved in redux (can check logs for googleSignInCompleted event or look in redux debugger in flipper)

iOS

https://github.com/valora-inc/wallet/assets/5062591/e432cac0-9992-4a03-94f7-3af7e12f777f


Android

https://github.com/valora-inc/wallet/assets/5062591/4c7874f4-8a7d-4909-9b31-ac53e8e0c062



### Related issues

- Fixes ACT-743

### Backwards compatibility

Yes
